src/nvim/highlight.c

```diff
@@ -375,6 +375,11 @@ void update_window_hl(win_T *wp, bool invalid)
       }
       wp->w_float_config.border_attr[i] = attr;
     }
+
+    if (wp->w_float_config.title_chunks.size == 1) {
+      int attr = win_hl_attr(wp, HLF_BTITLE);
+      wp->w_float_config.title_attr = attr;
+    }
```

Comment:
this is better done directly in parse_border_title. use `syn_check_group(S_LEN("FloatBorderTitle"))` to get it as an `hl_id`, then there is no special case when redrawing either.